### DISCUSSION TITLE
Rookies wanted!

### DISCUSSION THREAD

**Discussion by nomakewan** [gpstar81/GPStar-proton-pack#398] [General]
Okay, so as the minor hiccup in 5.4.0's release showed, there is only so much that three guys nose-deep in development can test. For example, my bench setup used 9-LED pucks and I tested the code for the 3-LED and 5-LED pucks on those, which worked fine. But it turned out there was an actual hardware difference on the 5-LED pucks causing a write speed issue. And with the Attenuator, Dustin and I both tested the new build on our ESP32s and had it working, yet when it hit the public several folks lost web access to their packs temporarily.

So that's where you, our lovely community, come in. Just like the new hire in the 2009 video game, we'd like a small, dedicated pool of beta testers to try out new changes before they get a seal of approval and pushed out to the community as a whole. So, what are we looking for? First, you must be willing to flash your firmware and update your SD cards readily. If updating is a chore for you, this is not for you. Next, we have specific categories for mods that we'd like to make sure we cover. See below.

# PACK GROUP
1. GPStar Lite. Bonus points if you have a wireless module or Attenuator!
2. Stock HasLab. This just means no LED changes to the powercell or outer cyclotron; any other changes are fine.
3. Frutto 20-LED. Must have the Frutto 20-LED setup installed on the outer cyclotron.
4. Frutto 36-LED. Must have the Frutto Max 36-LED setup installed on the outer cyclotron.
5. 40-LED NeoPixel Ring. Must be using a 40-LED ring for the outer cyclotron (Spirit packs welcome!).

Within the above, we would like if at least one person had a 12-LED inner cyclotron, a 23/24/26-LED inner cyclotron, and a 35/36-LED inner cyclotron. Likewise it would be nice to have at least one person using discrete LEDs for the switch panel, and at least one person with the Frutto addressable switch panel. At least one individual should have an N-Filter smoke kit (any kind), at least one individual should have a Booster Tube smoke kit (any kind). For the Pack Group, DIY builds are welcome as long as the meet the above criteria.

# WAND GROUP
1. Stock Hasbro w/ GPStar wand board. Barrel and bargraph must be stock. Can be standalone or used with pack.
2. Stock Hasbro barrel w/ bargraph upgrades. Can have any electronics upgraded as long as the barrel remains stock.
3. Frutto barrel. Can have any upgrades as long as the Frutto barrel is in use.

For the above, either 1 or 2 should be using a white barrel tip LED. At least one person should have add-on hat lights installed. For the Wand Group as a whole, DIY builds are welcome as long as they are using the Mega 2560 Pro Mini and not the old Nano build.

# ATTENUATOR GROUP
I will leave it to Dustin to define the parameters of what he's looking for in testers for the Attenuator.

**Comment by justindunn** [comment]
Hello, I am a software engineer and a huge ghostbusters fan, I am reverse engineering your software now to learn it, I am ready and willingt to help, reach out to me at axeouse@gmail.com
![75276746396__ECE297C2-9EC3-422A-BC90-61E2047DDA57](https://github.com/user-attachments/assets/653b5d60-5a37-4763-a028-13ca6565e0cf)

**Reply by JMills75** [reply]
I'd like to help.  I've been a dev for multiple years and now lead a team of software testers.  I fall in PACK GROUP 4 and WAND GROUP 3.  I have an ESP32, but no attenuator.

Aside from actually flashing and testing changes, I could help document a reusable test plan if you are interested.
